activerecord/lib/active_record/connection_adapters/sqlite3/schema_statements.rb

```diff
@@ -82,6 +82,10 @@ def remove_foreign_key(from_table, to_table = nil, **options)
           alter_table(from_table, foreign_keys)
         end
 
+        def virtual_table_exists?(table_name)
+          query_values(data_source_sql(table_name, type: "VIRTUAL TABLE"), "SCHEMA").any? if table_name.present?
```

Comment:
Do we need to check if `table_name.present?`? We always pass a table name for this method. If users pass `""` I'd just allow the query fail on sqlite.